xarray/core/utils.py

```diff
@@ -572,14 +578,14 @@ def size(self: Any) -> int:
         """
         return math.prod(self.shape)
 
-    def __len__(self: Any) -> int:
+    def __len__(self) -> int:
         try:
             return self.shape[0]
         except IndexError:
             raise TypeError("len() of unsized object")
 
 
-class NDArrayMixin(NdimSizeLenMixin):
+class NDArrayMixin(NdimSizeLenMixin, metaclass=ABCMeta):
```

Comment:
```suggestion
class NDArrayMixin(NdimSizeLenMixin, ABC):
```
Not sure about the order here. Maybe it's the opposite?